# PHPSTAN Result
```::error file=src/helper/redirect.php,line=14,col=0::Function redirect() has parameter $old_POST with no value type specified in iterable type array.
 ------ -------------------------------------------------------------------------------------- 
  Line   src/helper/redirect.php                                                               
 ------ -------------------------------------------------------------------------------------- 
  14     Function redirect() has parameter $old_POST with no value type                        
         specified in iterable type array.                                                     
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
 ------ -------------------------------------------------------------------------------------- 


 [ERROR] Found 1 error                                                          
```

--- a/src/helper/redirect.php
+++ b/src/helper/redirect.php
@@ -14,9 +14,9 @@
 function redirect(string $headerstring, int $response_code, array $old_POST = null, bool $replace = true): void
 {
 
-        if($old_POST != null) {
-                $_SESSION["old_POST"] = $old_POST;
-        }
+    if ($old_POST != null) {
+            $_SESSION["old_POST"] = $old_POST;
+    }
         header("location: {$headerstring}", $replace, $response_code);
 }
 ?>

--- a/src/models/User.php
+++ b/src/models/User.php
@@ -53,7 +53,7 @@ class User extends BaseModel
             session_unset();
             session_start();
             $_SESSION['message'] = $exception->getMessage();
-            redirect('/login',302,$_POST);
+            redirect('/login', 302, $_POST);
         }
     }
 
@@ -79,18 +79,18 @@ class User extends BaseModel
                 if ($result == 1) {
                  //   session_start();
                     $_SESSION['message'] = "Email bereits vergeben";
-                    redirect('/register',302, $_POST);
+                    redirect('/register', 302, $_POST);
                 } else {
                     $hashedPassword = password_hash($password, PASSWORD_DEFAULT);
                     $query = "Insert INTO users (forename,surname,password,email) values ('{$forename}','{$surname}','{$hashedPassword}','{$email}')";
                     $saved = $sql = $this->connection->query($query);
                     if (!$saved) {
                         $_SESSION['message'] = "Hoppla, da ist etwas schiefgelaufen";
-                        redirect('/register',302, $_POST);
+                        redirect('/register', 302, $_POST);
                     }
                     $userId = $this->connection->insert_id; // get id after creation
                     $_SESSION['user'] = $userId; // login aver successful creation
-                    redirect('/dashboard',302, $_POST);
+                    redirect('/dashboard', 302, $_POST);
                 }
             }
         } catch (Exception $e) {

--- a/src/views/partials/footer.view.php
+++ b/src/views/partials/footer.view.php
@@ -2,10 +2,10 @@
 </html>
 <?php
 // unset possible messages
-if(isset($_SESSION['old_POST'])){
+if (isset($_SESSION['old_POST'])) {
     unset($_SESSION["old_POST"]);
 }
-if(isset($_SESSION['message'])){
+if (isset($_SESSION['message'])) {
     unset($_SESSION['message']);
 }
 ?>

